src/dune_engine/action_exec.mli

```diff
@@ -27,12 +27,33 @@ module Exec_result : sig
   type t = { dynamic_deps_stages : (Dynamic_dep.Set.t * Dep.Facts.t) List.t }
 end
 
+(** [root] should be the root of the current build context, or the root of the
+    sandbox if the action is sandboxed. *)
 val exec :
      targets:Path.Build.Set.t
+  -> root:Path.t
   -> context:Build_context.t option
   -> env:Env.t
   -> rule_loc:Loc.t
   -> build_deps:(Dep.Set.t -> Dep.Fact.t Dep.Map.t Fiber.t)
   -> execution_parameters:Execution_parameters.t
   -> Action.t
   -> Exec_result.t Fiber.t
+
+(** The [BUILD_PATH_PREFIX_MAP] variable *)
+val _BUILD_PATH_PREFIX_MAP : string
+
+(** [extend_build_path_prefix_map env how map] extends the path rewriting rules
+    encoded in the [BUILD_PATH_PREFIX_MAP] variable.
+
+    Note that the rewriting rules are applied from right to left, so the last
+    rule of [map] will be tried first.
+
+    If the environment variable is already defined in [env], [how] explains
+    whether the rules in [map] should be tried before or after the existing
+    ones. *)
+val extend_build_path_prefix_map :
+     Env.t
+  -> [ `Existing_rules_have_precedence | `New_rules_have_precedence ]
```

Comment:
Polymorphic variants are back "in". Nice :)

Reply:
Yeah, I gave in

Reply:
Yeah, I followed the lead and sneaked in a polymorphic variant too :)

https://github.com/ocaml/dune/blob/65404cf973aa7ffc0e9dd7d05c9dd3709c7db2d4/otherlibs/stdune-unstable/temp.ml#L14-L20